### PR TITLE
SC8515/multi-db setups

### PR DESCRIPTION
**Pull request by andriitserkovnyi**
## PR Description

Documentation for https://spryker.atlassian.net/browse/SC-8515

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/cloud/dev/spryker-cloud-commerce-os/adding-databases-for-stores.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+regions:
+    EU:
+        services:
+            mail:
+                sender:
+                    name: Spryker No-Reply
+                    email: no-reply@spryker.local
+            databases:
+                eu-region-existing-database:
```
Comment (alexanderM91) [inline docs/cloud/dev/spryker-cloud-commerce-os/adding-databases-for-stores.md:24]: Why the database name contains `existing`?
Maybe it's better to name them:
database-de and database-at?

File: docs/cloud/dev/spryker-cloud-commerce-os/adding-databases-for-stores.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                        namespace: 2
+                    search:
+                        namespace: at_search
+                    database:
+                        name: eu-region-new-database
+```                        
+
+## Deploy the database
+
+Run a destructive pipeline for the application. Based on the environment, follow the instructions in one of the following docs:
```
Comment (alexanderM91) [inline docs/cloud/dev/spryker-cloud-commerce-os/adding-databases-for-stores.md:62]: Should we cover local development here?

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+Currently, the following setups are available:
+
+* Separated: a codebase with dedicated databases per store.
+* Shared: a codebase with all the stores sharing a database.
+
+
+## Shared setup
+
+With the shared setup, all the stores share a single database and codebase.
```
Comment (alexanderM91) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:17]: Share single database per region as it's possible to have EU and US regions, at least for local development

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+* Insignificant differences are covered in the code, for example, using code-buckets.
+
+* Stores are scaled together and don't need different deployment workflows.
+
+### Shared setup: Advantages
+
+The shared setup features:
+
+* Products, customers, orders, and so on are stored in the same database, which simplifies collaborative management.
```
Comment (alexanderM91) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:40]: It's possible to have fully separated data set per each store, like Products, customers, orders, and so on, which simplifies collaborative management?

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+* All the stores are deployed together, which is the most cost-effective solution.
```
Comment (alexanderM91) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:42]: What do you mean here?

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md (hunk 4 above)
Comment (andriitserkovnyi) [reply docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:40]: Are you sure this is applicable to the _shared_ setup?

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+* The stores are completely different from the perspective of the following:
+
+    * Design
+
+    * Business logic
+
+    * Features or modules
+
+* Independent maintenance and development: each store is handled by a dedicated team with its development workflow and release cycles.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:75]: -

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+* Several CI/CD(continuous integration/continuous delivery) pipelines are shipped for every store by default.
+
+* Better control over expenses of each store.
+
+* Flexible URL management. For example, the same product in different stores can have the same URL.
+
+* Flexible management of the configuration of stores: distinct category navigation, product schema details, and users.
+
+* Separate deployment of each store: deploy a new version of a store without affecting other stores.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:95]: Separate deployment of stores' databases: deploy a new version of a store's database without affecting the other databases. Deploy new stores without affecting the existing ones.

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+### Separated setup: Advantages
+
+
+The separated setup features:
+
+* Flexible deployment and scaling: since stores are independent of one another, deploy, remove, and scale each store without affecting other stores.
+
+* Several CI/CD(continuous integration/continuous delivery) pipelines are shipped for every store by default.
+
+* Better control over expenses of each store.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:89]: *

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+### Shared setup: Advantages
+
+The shared setup features:
+
+* Products, customers, orders, and so on are stored in the same database, which simplifies collaborative management.
+
+* All stores are hosted in the same AWS region.
+
+* Shared traffic distribution for all stores using ALB+NLBs.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:44]: *

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+* SSL certificates are generated automatically or managed manually in AWS.
+
+    * SSL termination process is handled by ALB.
+
+    * We can assign multiple certificates of different domains to one ALB.
+
+* Single CI/CD pipeline is used for all the stores.
+
+* On-demand setup of any type of environment: test, staging, and so on.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:54]: *

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+* Shared traffic distribution for all stores using ALB+NLBs.
+
+* SSL certificates are generated automatically or managed manually in AWS.
+
+    * SSL termination process is handled by ALB.
+
+    * We can assign multiple certificates of different domains to one ALB.
+
+* Single CI/CD pipeline is used for all the stores.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:52]: *

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+* All stores are hosted in the same AWS region.
+
+* Shared traffic distribution for all stores using ALB+NLBs.
+
+* SSL certificates are generated automatically or managed manually in AWS.
+
+    * SSL termination process is handled by ALB.
+
+    * We can assign multiple certificates of different domains to one ALB.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:50]: *

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+* Products, customers, orders, and so on are stored in the same database, which simplifies collaborative management.
+
+* All stores are hosted in the same AWS region.
+
+* Shared traffic distribution for all stores using ALB+NLBs.
+
+* SSL certificates are generated automatically or managed manually in AWS.
+
+    * SSL termination process is handled by ALB.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:48]: *

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+
+The shared setup features:
+
+* Products, customers, orders, and so on are stored in the same database, which simplifies collaborative management.
+
+* All stores are hosted in the same AWS region.
+
+* Shared traffic distribution for all stores using ALB+NLBs.
+
+* SSL certificates are generated automatically or managed manually in AWS.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:46]: *

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+
+We recommend this setup for the stores that can be described as follows:
+
+* All the stores are united into a multi-store system.
+
+* All the stores follow the same business logic.
+
+* Insignificant differences are covered in the code, for example, using code-buckets.
+
+* Stores are scaled together and don't need different deployment workflows.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:34]: *

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+
+### Shared setup: When to use
+
+
+We recommend this setup for the stores that can be described as follows:
+
+* All the stores are united into a multi-store system.
+
+* All the stores follow the same business logic.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:30]: -

File: docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+
+* Insignificant differences are covered in the code, for example, using code-buckets.
```
Comment (andriitserkovnyi) [inline docs/cloud/dev/spryker-cloud-commerce-os/multi-store-setups.md:32]: -

**Review by VadymSachenko — APPROVED** [review]

**Pull request merged**